components/subject/Summary.tsx

```diff
@@ -0,0 +1,283 @@
+import { classNames, pluralize } from '@/lib/util'
+import { ToolsOzoneModerationDefs } from '@atproto/api'
+import {
+  ExclamationCircleIcon,
+  ExclamationTriangleIcon,
+  FlagIcon,
+  PencilSquareIcon,
+  ShieldExclamationIcon,
+  UserCircleIcon,
+} from '@heroicons/react/24/solid'
+import { ComponentProps, useState } from 'react'
+
+export const StatView = ({
+  count,
+  Icon,
+  appearance,
+  text,
+  title,
+  ...rest
+}: {
+  count: number
+  Icon: (props: React.ComponentProps<'svg'>) => JSX.Element
+  text?: string
+  appearance?: 'danger' | 'info' | 'warning' | 'success'
+} & ComponentProps<'button'>) => {
+  let bgColor = 'bg-gray-400 dark:text-gray-100 text-white'
+
+  if (appearance === 'info') {
+    bgColor = 'bg-blue-400 text-blue-800'
+  } else if (appearance === 'danger') {
+    bgColor = 'bg-red-200 text-red-800'
+  } else if (appearance === 'warning') {
+    bgColor = 'bg-orange-200 text-orange-800'
+  } else if (appearance === 'success') {
+    bgColor = 'bg-green-200 text-green-800'
+  }
+
+  return (
+    <button
+      type="button"
+      className={classNames(
+        'flex flex-row items-center gap-1 rounded px-1 text-xs',
+        bgColor,
+      )}
+      title={title}
+      aria-label={title}
+      {...rest}
+    >
+      <Icon className="w-3 h-3" />
+      {count}
+      {text && <span>{text}</span>}
+    </button>
+  )
+}
+
+export const AccountStats = ({
+  stats,
+  showAll,
+  onAccountTakedownClick,
+}: {
+  showAll: boolean
+  onAccountTakedownClick?: () => void
+  stats: ToolsOzoneModerationDefs.AccountStats
+}) => {
+  const { takedownCount, suspendCount, appealCount, reportCount } = stats
+  // Dont want to make the UI noisy with appeal count if we're in summary mode and already showing takedown and suspension
+  const shouldShowAppeal =
+    appealCount && (showAll || !takedownCount || !suspendCount)
+  // Dont want to make the UI noisy with report count if we're in summary mode and we have both takedown and suspension count
+  const shouldShowReport =
+    reportCount && (showAll || (!takedownCount && !suspendCount))
+  const hasStats = takedownCount || suspendCount || appealCount || reportCount
```

Comment:
IMO A `zero` should count as "has stats". But if that's really not what you want, this could should explicitly account for `0` vs `undefined`.

```suggestion
  const { takedownCount, suspendCount, appealCount, reportCount } = stats

    // Dont want to make the UI noisy with appeal count if we're in summary mode and already showing takedown and suspension
  const shouldShowAppeal = appealCount == null
    ? true // No data <== I suggest we *do* show that we don't have data
    : showAll ||
      !takedownCount || // No data or zero
      !suspendCount // No data or zero

  // Dont want to make the UI noisy with report count if we're in summary mode and we have both takedown and suspension count
  const shouldShowReport = reportCount == null
    ? true // no data <== 
    : showAll || (!takedownCount && !suspendCount)

  const hasStats =
    takedownCount != null ||
    suspendCount != null ||
    appealCount != null ||
    reportCount != null
```

Reply:
generally agreed but I don't think it impacts mod decisions to know that "we don't have stats" vs. the stats is 0 and I'm trying to be cautious about making the UI too noisy with stats.
at least to start with, I see these as additional metadata to help enforce decisions made from other factors already rather than a primary factor for making decision.